analyzers/src/SonarAnalyzer.VisualBasic/Rules/CertificateValidationCheck.cs

```diff
@@ -34,74 +34,66 @@ namespace SonarAnalyzer.Rules.VisualBasic
 {
     [DiagnosticAnalyzer(LanguageNames.VisualBasic)]
     [Rule(DiagnosticId)]
-    public sealed class CertificateValidationCheck : CertificateValidationCheckBase<MethodBlockSyntax, ArgumentSyntax, ExpressionSyntax, IdentifierNameSyntax, AssignmentStatementSyntax, InvocationExpressionSyntax, ParameterSyntax, ModifiedIdentifierSyntax, LambdaExpressionSyntax>
+    public sealed class CertificateValidationCheck : CertificateValidationCheckBase<
+        MethodBlockSyntax,
+        ArgumentSyntax,
+        ExpressionSyntax,
+        IdentifierNameSyntax,
+        AssignmentStatementSyntax,
+        InvocationExpressionSyntax,
+        ParameterSyntax,
+        ModifiedIdentifierSyntax,
+        LambdaExpressionSyntax,
+        MemberAccessExpressionSyntax>
     {
         public CertificateValidationCheck() : base(RspecStrings.ResourceManager) { }
 
+        internal override AbstractMethodParameterLookup<ArgumentSyntax> CreateParameterLookup(SyntaxNode argumentListNode, IMethodSymbol method) =>
+            argumentListNode switch
+            {
+                InvocationExpressionSyntax invocation => new VisualBasicMethodParameterLookup(invocation.ArgumentList, method),
+                ObjectCreationExpressionSyntax ctor => new VisualBasicMethodParameterLookup(ctor.ArgumentList, method),
+                _ => throw new ArgumentException("Unexpected type.", nameof(argumentListNode)) // This should be throw only by bad usage of this method, not by input dependency
+            };
+
         protected override void Initialize(SonarAnalysisContext context)
         {
-            //C# += equivalent:
-            //AddHandler situation does not exist in VB.NET. Delegate is pointer to one function only and can not have multiple handlers. It's not an event.
-            //Only assignment and object creation are valid cases for VB.NET
+            // C# += equivalent:
+            // AddHandler situation does not exist in VB.NET. Delegate is pointer to one function only and can not have multiple handlers. It's not an event.
+            // Only assignment and object creation are valid cases for VB.NET
```

Comment:
Is this comment really needed?
Actually wondering if any of these  handling of comments are needed in both c# and vbnet as the code itself is pretty self descriptive.

Reply:
These comments were there from original version. I added them back then because I was the only one knowing this out of the box. I find it still useful for non-VB experts :)

Reply:
..it explains why is the C# registration different from VB